core/client/src/main/java/alluxio/client/block/stream/LocalFilePacketWriter.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.stream;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.client.block.BlockWorkerClient;
+import alluxio.worker.block.io.LocalFileBlockWriter;
+
+import com.google.common.base.Preconditions;
+import io.netty.buffer.ByteBuf;
+
+import java.io.IOException;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A local packet writer that simply writes packets to a local file.
+ */
+@NotThreadSafe
+public final class LocalFilePacketWriter implements PacketWriter {
+  private static final long PACKET_SIZE =
+      Configuration.getBytes(PropertyKey.USER_LOCAL_WRITER_PACKET_SIZE_BYTES);
+  private static final long FILE_BUFFER_BYTES =
+      Configuration.getBytes(PropertyKey.USER_FILE_BUFFER_BYTES);
+
+  /** The position to write the next byte at. */
+  private long mPos = 0;
+  /** The number of bytes reserved on the block worker to hold the block. */
+  private long mPosReserved = 0;
```

Comment:
This comment is probably related to the one you made for LocalFilePacketWriter#reserve?
mPosReserved makes more sense here because reserve takes an absolute pos. And a block always starts with pos=0